### PR TITLE
Pull request re-triggering events

### PR DESCRIPTION
**Pull request by phracek**
Pull request adds support for re-triggering `/packit copr-build` and `/packit build`.
`/packit build <WITH ARGS>` will follow.

In case of `/packit build` failed, a message in Pull Request looks like https://github.com/packit-service/hello-world/pull/9#issuecomment-522545125

I have moved copr-build functionality into alone class, so we do not duplicate code for `GithubCoprBuildHandler` (https://github.com/packit-service/packit-service/blob/master/packit_service/worker/github_handlers.py#L212) and `GitHubPullRequestCommentCoprBuildHandler` which is introduced by this PR.
Testing JSON data are also part of this pull request.

Also in case copr-build failed, following message is shown https://github.com/packit-service/packit-service/compare/master...phracek:pull_request_events_handler?expand=1#diff-b763b88aebc07acd21bb55e2459d07f7R105

Pull request covers:
- [x] re-triggering COPR build by PR comment `/packit copr-build`
- [x] tests are covered

### CODE REVIEW

File: packit_service/worker/jobs.py [hunk 1]
```diff
@@ -97,6 +102,47 @@ def process_jobs(self, event: Optional[Any]) -> Dict[str, HandlerResults]:
                     handler.clean()
         return handlers_results
 
+    def process_comment_jobs(self, event: Optional[Any]):
+        handlers_results = {}
+        # packit_command can be `/packit build` or `/packit build <with_args>`
+        (packit_mark, *packit_command) = event.comment.split(maxsplit=3)
```
Comment (TomasTomecek) [inline packit_service/worker/jobs.py]: I'd catch ValueError here, just in case:
```
In [8]: a, *b = "".split(maxsplit=3)       
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-757e103c7539> in <module>
----> 1 a, *b = "".split(maxsplit=3)
        global a = 'a'
        global b.split = undefined
        global maxsplit = undefined

ValueError: not enough values to unpack (expected at least 1, got 0)
```
I guess that empty comments can't happen, but let's be prepared.

File: packit_service/worker/jobs.py [hunk 2]
```diff
@@ -97,6 +102,47 @@ def process_jobs(self, event: Optional[Any]) -> Dict[str, HandlerResults]:
...
+
+        if REQUESTED_PULL_REQUEST_COMMENT != packit_mark:
+            logger.debug(
+                f"This PR comment '{packit_mark}' is not handled by packit-service"
+            )
+            return HandlerResults(success=False, details={})
+
+        # packit has command `copr-build`. But PullRequestCommentAction has enum `copr_build`.
+        packit_action = PullRequestCommentAction[packit_command[0].replace("-", "_")]
```
Comment (TomasTomecek) [inline packit_service/worker/jobs.py]: a comment `/packit` will traceback here on `IndexError`

```
In [1]: a, *b = "/packit".split(maxsplit=3)

In [2]: b[0]                               
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-2-54af0eb47770> in <module>
----> 1 b[0]
        global b = []

IndexError: list index out of range
```

File: packit_service/worker/jobs.py [hunk 3]
```diff
@@ -97,6 +102,47 @@ def process_jobs(self, event: Optional[Any]) -> Dict[str, HandlerResults]:
...
+        handler_kls: Any = PULL_REQUEST_COMMENT_HANDLER_MAPPING.get(packit_action, None)
+        if not handler_kls:
+            return HandlerResults(
+                success=False,
+                details={
+                    "msg": (
+                        f"This PR trigger command '{REQUESTED_PULL_REQUEST_COMMENT} "
```
Comment (TomasTomecek) [inline packit_service/worker/jobs.py]: this message would be slightly misleading since this event will be triggered for *all* PR comments, even those which are not commands for bots, but text for humans, I'd probably rephrase this to
```python
f"PR comment '{event.comment[:35]}' is not a packit-service command"
```

first 35 chars should be enough to store, we don't need complete comments

File: packit_service/worker/github_handlers.py [hunk 4]
```diff
@@ -517,3 +460,130 @@ def run(self) -> HandlerResults:
...
+                return job
+        return None
+
+    def run(self) -> HandlerResults:
+
+        collaborators = self.project.who_can_merge_pr()
+        r = BuildStatusReporter(self.project, self.event.commit_sha)
+        if self.event.github_login not in collaborators:
+            msg = "Only collaborators can trigger Packit-as-a-Service"
+            r.set_status("failure", msg, PRCheckName.get_build_check())
```
Comment (TomasTomecek) [inline packit_service/worker/github_handlers.py]: Here's a scenario:
1. I create a PR on a project where I don't have commit rights
2. Someone with commit access types `/packit copr-build`
3. Build passes, check is now green
4. I type `/packit copr-build`
5. Packit service sets the check to "Only collaborators can trigger Packit-as-a-Service"

Is this a correct assumption? If so, in such a case, it would be better for p-s to comment on the PR instead of setting the status.

File: packit_service/worker/github_handlers.py [hunk 5]
```diff
@@ -517,3 +460,130 @@ def run(self) -> HandlerResults:
...
+                )
+                testing_farm_handler.run()
+            else:
+                logger.debug("Testing farm not in the job config.")
+            return HandlerResults(success=True, details={})
+        return handler_results
+
+
+@add_to_pr_comment_mapping
+class GitHubPullRequestCommentBuildHandler(PullRequestCommentHandler):
```
Comment (TomasTomecek) [inline packit_service/worker/github_handlers.py]: Now that I look at this, I probably don't follow the E2E workflow: should people type `/packit build` to get a prod koji build of HEAD of dist-git master? My concern is that the PR is completely irrelevant since it's not used during the build in any way.

File: packit_service/worker/jobs.py (hunk 1 above)
Comment (phracek) [reply packit_service/worker/jobs.py]: Fixed by https://github.com/packit-service/packit-service/pull/89/commits/639e77d7449b0945c94aa5ef786207cda6f66db3

File: packit_service/worker/github_handlers.py (hunk 4 above)
Comment (phracek) [reply packit_service/worker/github_handlers.py]: Yes, your assumption is right. I will add a PR comment instead of setting status.

File: packit_service/worker/github_handlers.py [hunk 6]
```diff
@@ -242,97 +251,22 @@ def __init__(
 
     def handle_pull_request(self):
 
-        check_name = PRCheckName.get_build_check()
```
[the remark this answers is not in the document]
Comment (phracek) [reply packit_service/worker/github_handlers.py]: It was cased by rebasing from `master`. Too many changes after my vacation. Sorry

File: packit_service/worker/github_handlers.py [hunk 7]
```diff
@@ -242,97 +251,22 @@ def __init__(
...
-            project=project, owner=owner, chroots=chroots
-        )
-        r = BuildStatusReporter(self.project, self.event.commit_sha, copr_build_model)
+        r = BuildStatusReporter(self.project, self.event.commit_sha)
         if self.event.github_login not in collaborators:
             msg = "Only collaborators can trigger Packit-as-a-Service"
             r.set_status("failure", msg, PRCheckName.get_build_check())
             return HandlerResults(success=False, details={"msg": msg})
-        try:
-            r.report("pending", "RPM build has just started...", check_name=check_name)
```
[the remark this answers is not in the document]
Comment (phracek) [reply packit_service/worker/github_handlers.py]: Moved to CoprBuildHandler class.

File: packit_service/worker/github_handlers.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (phracek) [reply packit_service/worker/github_handlers.py]: Fixed

File: packit_service/worker/github_handlers.py (hunk 5 above)
Comment (phracek) [reply packit_service/worker/github_handlers.py]: The code is commented out.

Comment (jpopelka) [reply packit_service/worker/github_handlers.py]: I don't follow the conversation/PR to a detail (due to it's being too huge), but commenting out such a big parts of code usually isn't a good solution.

Comment (phracek) [reply packit_service/worker/github_handlers.py]: Yesterday, we decided to comment out on grooming.
